apps/eth-wrapper/src/components/Wrapper.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useEffect, useState, useMemo } from 'react';
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import { SafeAppsSdkProvider } from '@gnosis.pm/safe-apps-ethers-provider';
+import { Snackbar } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+import { Button, TextField } from '@gnosis.pm/safe-react-components';
+import { getWethAddress, Erc20 } from '../utils/Erc20Constants';
+import { ethers } from 'ethers';
+import { WETHwithdraw_function } from '../utils/WETHConstants';
+
+interface WrapperProps {
+    wrap: boolean
+}
+
+const Wrapper: React.FC<WrapperProps> = (props: WrapperProps) => {
+    const { sdk, safe } = useSafeAppsSDK();
+
+    const [amountToWrap, setAmountToWrap] = useState("");
+    const [errorMessage, setErrorMessage] = useState("");
+    const [availableBalance, setAvailableBalance] = useState(0.0);
+    const [isError, setIsError] = useState(false);
+    const [safeTxHash, setSafeTxHash] = useState("");
+
+    const provider = useMemo(() => new SafeAppsSdkProvider(safe, sdk), [safe, sdk]);
+    const weth = useMemo(() => new ethers.Contract(getWethAddress(safe.network.toLowerCase()), Erc20, provider), [provider, safe]);
```

Comment:
It might be better to create the contract first without a provider and then just connect to it. This will minimize the creation overhead (same for the WethBalance component).